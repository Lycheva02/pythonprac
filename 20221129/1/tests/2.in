class C(metaclass=dump):
    def __init__(self, s):
	    self.s = s[-len(s)//2:] + s[:len(s)//2]
    def reverse(self, reset=False):
        if reset:
            self.s = self.s[::-1]
            return self.s
        return self.s[::-1]
		
c = C('ogogoqkrq')
d = C('aaabbb')
print(c.reverse(), d.reverse(reset=True))