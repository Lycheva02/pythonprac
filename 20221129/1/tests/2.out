__init__: ('ogogoqkrq',), {}
__init__: ('aaabbb',), {}
reverse: (), {}
reverse: (), {'reset': True}
gogoqrkqo aaabbb
